src/core/gltf2exporter/dracocompressor_p.cpp

```diff
@@ -0,0 +1,312 @@
+                       attribute.vertexSize() * sizeof(T),
+                       0);
+    auto attId = mesh.AddAttribute(meshAttribute, true, attribute.count());
+    if (attId == -1)
+        return { {}, -1 };
+
+    for (unsigned int i = 0; i < attribute.count(); ++i)
+        mesh.attribute(attId)->SetAttributeValue(draco::AttributeValueIndex(i), &packedData.data()[i * attribute.vertexSize() * sizeof(T)]);
+
+    return { attribute.property("semanticName").toString(), attId };
+}
+
+template<typename T>
+QVector<T> createIndicesVectorFromAttribute(const Qt3DRender::QAttribute &attribute)
+{
+    const auto byteStride = attribute.byteStride();
+    const auto byteOffset = attribute.byteOffset();
+    const auto data = attribute.buffer()->data().mid(byteOffset);
+
+    QVector<T> indicesVector;
+    indicesVector.resize(attribute.count());
+
+    if (byteStride == 0)
+    {
+        for (int i = 0, nbIndices = indicesVector.size(); i < nbIndices; ++i)
+            indicesVector[i] = *reinterpret_cast<const T*>(data.data() + i * sizeof(T));
+    }
+    else
+    {
+        for (int i = 0, nbIndices = indicesVector.size(); i < nbIndices; ++i)
+            indicesVector[i] = *reinterpret_cast<const T*>(data.data() + i * byteStride);
+    }
+
+    return indicesVector;
+}
+
+template<typename T>
+void addFacesToMesh(const QVector<T> &indices, draco::Mesh &dracoMesh)
+{
+    for (int i = 0, nbIndices = indices.size() / 3; i < nbIndices; ++i)
+        dracoMesh.AddFace({ static_cast<draco::IndexType<unsigned int, draco::PointIndex_tag_type_>>(indices[3 * i + 0]),
+                            static_cast<draco::IndexType<unsigned int, draco::PointIndex_tag_type_>>(indices[3 * i + 1]),
+                            static_cast<draco::IndexType<unsigned int, draco::PointIndex_tag_type_>>(indices[3 * i + 2]) });
+}
+} // namespace
+
+CompressedMesh compressMesh(
+        const Qt3DRender::QGeometry &geometry,
+        const GLTF2ExportConfiguration& conf
+        )
+{
+    std::unique_ptr<draco::EncoderBuffer> compressBuffer(new draco::EncoderBuffer);
+    std::vector<std::pair<QString, int>> attributes;
+    draco::Mesh dracoMesh;
+
+    std::unordered_map<Qt3DRender::QBuffer *, draco::DataBuffer *> attributeBufferToDataBuffer;
+    // Convert Qt3D geometry to draco mesh
+    const auto geometryAttributes = geometry.attributes();
+    Qt3DRender::QAttribute *indicesAttribute = nullptr;
+    for (const auto attribute : geometryAttributes) {
+        if (attribute->name() == Qt3DRender::QAttribute::defaultPositionAttributeName())
+            dracoMesh.set_num_points(attribute->count());
+    }
+
+    for (const auto attribute : geometryAttributes) {
+        if (attribute->attributeType() == Qt3DRender::QAttribute::IndexAttribute) {
+            indicesAttribute = attribute;
+            continue;
+        }
+
+        qDebug() << "compressing: " << attribute->name() << attribute->vertexBaseType() << attribute->attributeType();
```

Comment:
maybe use categorized logging instead?

Reply:
I've removed this qdebug, it was only to help me fix a bug - what kind of useful logging could be added instead ?

Reply:
You could keep it but maybe log it into a dedicated category Kuesa::DracoCompressor or something?

Reply:
done